util/pkg/reflectutils/access.go

```diff
@@ -45,6 +47,7 @@ func SetString(target interface{}, targetPath string, newValue string) error {
 				return fmt.Errorf("cannot set field %q: %v", path, err)
 			}
 
+			fieldSet = true
```

Comment:
Seems that this is the only place where this is set to `true` and the function exists immediately after. Maybe I am missing something?

Reply:
Yup. So the way this works as per my understanding is that, we recurse into the structure searching for the targetFieldPath. We set the targetFieldPath to the given value if we find an exact match in the structure. If during searching we see that the struct is `nil` i.e if we are setting `spec.networking.cilium.prometheusPort` but `spec.networking.cilium == nil` then it will autopopulate the structure value with `spec.networking.cilium` . The recursion begins again and we end up finding an exact match after which we set the field to the value. 

In the case of a field not being present, we ll never get a match and just recurse the entire struct and exit silently.

Reply:
Ah, missed the recursive function part. 😄